feature/tunnel/otg_tests/tunnel_interface_based_resize_test/tunnel_interface_based_resize_test.go

```diff
@@ -0,0 +1,520 @@
+	config := &oc.Root{}
+
+	dp1 := dut.Port(t, "port1")
+	ingressp := ingressIntf.ConfigOCInterface(config.GetOrCreateInterface(dp1.Name()), dut)
+	gnmi.Replace(t, dut, gnmi.OC().Interface(ingressp.GetName()).Config(), ingressp)
+
+	dp6 := dut.Port(t, "port6")
+	egressp := egressIntf.ConfigOCInterface(config.GetOrCreateInterface(dp6.Name()), dut)
+	gnmi.Replace(t, dut, gnmi.OC().Interface(egressp.GetName()).Config(), egressp)
+
+	aggID = netutil.NextAggregateInterface(t, dut)
+	configureInterfaces(t, dut, egressLag1Ports, []*attrs.Attributes{&egressLag1Intf}, aggID)
+	aggID = netutil.NextAggregateInterface(t, dut)
+	configureInterfaces(t, dut, egressLag2Ports, []*attrs.Attributes{&egressLag2Intf}, aggID)
+	configureStaticRoutes(t, dut)
+	_, ni, pf := cfgplugins.SetupPolicyForwardingInfraOC(ocPFParams.NetworkInstanceName)
+	EncapInGRE(t, dut, pf, ni, ocPFParams, ocNHGParams)
+
+}
+
+func TestSetup(t *testing.T) {
+	t.Log("TUN:1.6.1-Setup Tunnels as per requirements")
+	dut := ondatra.DUT(t, "dut")
+	fptest.ConfigureDefaultNetworkInstance(t, dut)
+
+	// Get default parameters for OC Policy Forwarding
+	ocPFParams := GetDefaultOcPolicyForwardingParams(t, dut)
+	ocNHGParams := GetDefaultStaticNextHopGroupParams()
+
+	// Pass ocPFParams to ConfigureDut
+	ConfigureDut(t, dut, ocPFParams, ocNHGParams)
+	ConfigureOTG(t)
+}
+
+func TestTunnelBaselineStats(t *testing.T) {
+	t.Log("TUN:1.6.2-Gather baseline stats by passing traffic")
+	ate := ondatra.ATE(t, "ate")
+	createflow(t, top, FlowIPv4, true)
+	createflow(t, top, FlowIPv6, false)
+	sendTraffic(t, ate)
+	time.Sleep(20 * time.Second)
+	ate.OTG().StopTraffic(t)
+	if err := FlowIPv4Validation.ValidateLossOnFlows(t, ate); err != nil {
+		t.Errorf("ValidateLossOnFlows(): got err: %q, want nil", err)
+	}
+	if err := FlowIPv6Validation.ValidateLossOnFlows(t, ate); err != nil {
+		t.Errorf("ValidateLossOnFlows(): got err: %q, want nil", err)
+	}
+	if err := ECMPValidation.ValidateECMP(t, ate); err != nil {
+		t.Errorf("ECMPValidationFailed(): got err: %q, want nil", err)
+	}
+
+}
+
+func TestTunnelInterfaceBasedResize(t *testing.T) {
+	t.Log("TUN:1.6.3-Removal of static routes to prevent using some of the tunnels")
+	ate := ondatra.ATE(t, "ate")
+	dut := ondatra.DUT(t, "dut")
+	sendTraffic(t, ate)
+	t.Log("Reduce number of Tunnels by deleting static routes")
+	deleteStaticRoutes(t, dut)
+
+	// Verify the entry for static route is not active through AFT Telemetry.
+	ipv4Path := gnmi.OC().NetworkInstance(deviations.DefaultNetworkInstance(dut)).Afts().Ipv4Entry(fmt.Sprintf("%s/24", staticRoute))
+	if got, ok := gnmi.Watch(t, dut, ipv4Path.State(), time.Minute, func(val *ygnmi.Value[*oc.NetworkInstance_Afts_Ipv4Entry]) bool {
+		ipv4Entry, present := val.Val()
+		t.Log(ipv4Entry.GetPrefix())
+		return present == false && ipv4Entry.GetPrefix() != fmt.Sprintf("%s/24", staticRoute)
```

Comment:
Will this cause a panic if the route is removed and present == false? Wouldn't ipv4Entry be nil at that point?